### PR TITLE
[Search] "Launch" Beta

### LINKED ISSUE #6720: "Launch" Search Beta Release
## What / Why

Lots of great stuff in #6569 and #6682 has effectively made production-ready search engines available.  These are still a little hidden from adopters, however, because they are not yet easily discoverable.

In order to "launch" the search beta, we need to make these things default / discoverable / clearly documented.

## Scope / UAT Target
- [x] As an App Integrator, my created app's SearchPage implementation should account for pagination (depends on #6062)
- [x] As an App Integrator, my created app should include the `DefaultTechDocsCollator` in the backend, and an updated Search Page frontend that includes the `techdocs` type (not directly related to the beta milestone, but we get questions about this often)
- [x] As an App Integrator, if I select `postgres` as my data store when running `create-app`, my Backstage search should default to using the `PgSearchEngine` implementation instead of `LunrSearchEngine`.

### PR DESCRIPTION
**Pull request by iamEAP**
## Hey, I just made a Pull Request!

Some small documentation and discoverability updates are all that's left from our original goals for a beta-level search platform.

Closes #6720 

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by iamEAP** [comment]
😅  Didn't expect those `create-app` and `e2e-test` changes to _just work_ on the first go.

**Review by camilaibs — APPROVED** [review]

File: packages/create-app/templates/default-app/packages/backend/src/plugins/search.ts.hbs [hunk 1]
```diff
@@ -16,7 +19,14 @@ export default async function createPlugin({
   tokenManager,
 }: PluginEnvironment) {
   // Initialize a connection to a search engine.
+  {{#if dbTypeSqlite}}
   const searchEngine = new LunrSearchEngine({ logger });
+  {{/if}}
+  {{#if dbTypePG}}
+  const searchEngine = (await PgSearchEngine.supported(database))
+    ? await PgSearchEngine.from({ database })
+    : new LunrSearchEngine({ logger });
```
Comment (Rugvip) [inline packages/create-app/templates/default-app/packages/backend/src/plugins/search.ts.hbs:28]: Kinda wondering if we should simply just always install it with the fallback instead? It's not like there are any other DB options anyway x)

**Review by Rugvip — APPROVED** [review]
Looks good, with some thoughts on the template

File: packages/create-app/templates/default-app/packages/backend/src/plugins/search.ts.hbs (hunk 1 above)
Comment (iamEAP) [reply packages/create-app/templates/default-app/packages/backend/src/plugins/search.ts.hbs:28]: Yeah, I had similar thoughts while doing this.  My thinking is: PG search will likely fulfill the needs of a certain niche of adopters, but that it's probably not the right choice for many or perhaps most, long-term.  So we don't want to encourage it _that_ much.  But respecting the `create-app` flag seems reasonable?  🤷‍♂️ 

We can tweak again based on feedback.

**Pull request merged**